client/blocks/signup-form/index.jsx

```diff
@@ -836,7 +839,21 @@ class SignupForm extends Component {
 						{
 							args: { email: userExistsError.email },
 							components: {
-								a: <a href={ this.getLoginLink( { emailAddress: userExistsError.email } ) } />,
+								a: (
+									<a
+										href={ loginLink }
+										onClick={ ( event ) => {
+											event.preventDefault();
+											this.props.handoffConnectSocialUser( userExistsError.email, {
+												service: this.props.step?.service,
+												access_token: this.props.step?.access_token,
+												id_token: this.props.step?.id_token,
+											} );
+											const { pathname, search } = getUrlParts( loginLink );
+											page( pathname + search );
```

Comment:
Next steps:

- [ ] Include `service`, `access_token`, `id_token` as query params.
- [ ] Inside `enhanceContextWithLogin`, read data from the query params and set it directly in the store.
- [ ] Remove the query params from browser history once the page loads.